### DISCUSSION TITLE
How to transcribe and translate at the same time to create two separate files?

### DISCUSSION THREAD

**Discussion by eric-bks** [openai/whisper#1500] [Q&A] [answered]
In Python, preferably.

I have extensive files and don't want to wait twice as long to run the same command. How do I get Korean and English SRT files from my function?

Here's what I usually use:
``` python3
def transcribe():
    model = whisper.load_model("large", device = 'cuda')
    audio = whisper.load_audio(f'{output_name}.mp4')
    global result
    result = model.transcribe(audio, verbose = True, language = 'Korean', task = 'translate')
```

I've tried this: `model.transcribe(audio, verbose = True, language = 'Korean', task = 'transcribe' + 'translate')`, but that doesn't work, and I imagine it would still take as much time as running it separately.

**Comment by phineas-pta** [comment]
u can open 2 different python process, 1 for transcribe 1 for translate

**Reply by eric-bks** [reply]
I still share resources, so they would both run half as fast, and this wouldn't save time.

Doesn't whisper have to transcribe first before it can translate? So I imagine that the transcription is already created when performing the translation. I should be able to output both from one operation.

**Reply by phineas-pta** [reply, marked as answer]
> Doesn't whisper have to transcribe first before it can translate?

no, whisper was trained to output directly translation from audio, u can read the paper

**Reply by spiteria** [reply]
@phineas-pta Where can I find the paper you are referring to?

**Reply by phineas-pta** [reply]
![image](https://github.com/openai/whisper/assets/37548991/5713b7f5-ec78-4bf2-a661-8ba7b75d2398)

**Comment by View-my-Git-Lab-krafi** [comment]
any update in one year
